server/src/controllers/course.ts

```diff
@@ -90,19 +101,105 @@ function parseSisuInstance(instance: SisuInstance): InstanceData {
   };
 }
 
+// Middleware function for handling errors in JSON parsing
+export function handleInvalidRequestJson(err: JsonError, req: Request, res: Response, next: NextFunction): void {
+  if (err instanceof SyntaxError && err.status === 400 && err.body !== undefined) {
+    res.status(400).send({
+      success: false,
+      errors: [ `SyntaxError: ${err.message}: ${err.body}` ]
+    });
+  } else
+    next();
+}
 
 export async function addCourse(req: Request, res: Response): Promise<void> {
+  const requestSchema: yup.AnyObjectSchema = yup.object().shape({
+    courseCode: yup.string().required(),
+    translations: yup.array().of(yup.object().shape({
```

Comment:
Since we're already returning the department and name in the `LocalizedString` style, we should make it so POST requests also require the same style:
```
{
  courseCode: 'ELEC-A7200',
  name: {
    en: 'Signals and Systems',
    fi: 'Signaalit ja järjestelmät',
    sv: ''
  },
  department: {
    en: 'School of Electrical Engineering',
    fi: 'Sähkötekniikan korkeakoulu',
    sv: ''
  }
};
```

Reply:
> Since we're already returning the department and name in the `LocalizedString` style, we should make it so POST requests also require the same style:
> 
> ```
> {
>   courseCode: 'ELEC-A7200',
>   name: {
>     en: 'Signals and Systems',
>     fi: 'Signaalit ja järjestelmät',
>     sv: ''
>   },
>   department: {
>     en: 'School of Electrical Engineering',
>     fi: 'Sähkötekniikan korkeakoulu',
>     sv: ''
>   }
> };
> ```

I was first considering this style in request. Then I thought that how to validate that the translation for one language is not missing a field. I found out that it would be a bit tricky to implement it with yup. Do you have suggestions?

If that kind of validation is not so critical, I can change the request style right away.

Reply:
Yup allows for defining custom validation methods.
https://github.com/jquense/yup#addmethodschematype-schema-name-string-method--schema-void

We could define a method for `yup.object` which checks for the fields of `LocalizedString`, this could then be used elsewhere too.

Reply:
I thought about this kind of scenario: _name_ and _department_ are translated into two languages, utilizing localized strings. If English translation of _department_ is missing but other English translations exist, fail the validation.

Maybe I'm just overthinking this.

Reply:
That might be reasonable but I'm not sure if it's necessary, there could also be a scenario such as:
```
{
  courseCode: 'CS-A1120',
  name: {
    en: 'Programming 2',
    fi: '',
    sv: ''
  },
  department: {
    en: 'Department of Computer Science',
    fi: 'Tietotekniikan laitos',
    sv: 'Institutionen för datateknik'
  }
};
```
I think that this is fine and should pass validation, though the Finnish and Swedish names are missing while the department exists.

Reply:
True

Reply:
Maybe a trivial question, but should we then require the field with empty string or allow it to be left undefined?

Reply:
I would say that it should still be defined to match the `LocalizedString` interface definition, but not sure if it matters.

Reply:
Implementing this so that undefined translations are replaced with empty string when inserting them into database. Yup schema for localized string checks that the object has at least one translation and no translations for any other languages except the ones supported.